### PR TITLE
feat(@desktop/wallet): keypair rename (a profile keypair name should follow display name)

### PR DESCRIPTION
**Pull request by saledjenic**
This PR includes the following:
- display name length limitation updated from 24 to 20 characters, keypair length is 20 characters as well and we cannot support more than that cause keycard allows only 20 characters for the keypair name @John-44  @benjthayer 
- menu added for all keypairs in wallet/settings part
- rename keypair feature implemented
- in case of a profile keypair, follows user's display name
- added to syncing (feature video, below)

Fixes: #10769

https://github.com/status-im/status-desktop/assets/86303051/945aa5e2-e519-4cff-b08f-a37f4b4d4575

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: ui/imports/utils/Constants.qml [hunk 1]
```diff
@@ -466,7 +466,28 @@ QtObject {
...
+                validate: function (t) { return !t.startsWith(" ") }
+                errorMessage: qsTr("Keypair starting with whitespace are not allowed")
+            },
+            StatusRegularExpressionValidator {
+                regularExpression: /^[a-zA-Z0-9\-_ ]+$/
+                errorMessage: qsTr("Only letters, numbers, underscores, whitespaces and hyphens allowed")
+            },
+            StatusMinLengthValidator {
+                minLength: keypair.nameLengthMin
+                errorMessage: qsTr("Keypair must be at least %1 characters").arg(keypair.nameLengthMin)
```
Comment (caybro) [inline ui/imports/utils/Constants.qml:487]:
```suggestion
                errorMessage: qsTr("Keypair must be at least %n character(s)", "", keypair.nameLengthMin)
```

File: ui/imports/utils/Constants.qml [hunk 2]
```diff
@@ -466,7 +466,28 @@ QtObject {
...
+
     readonly property QtObject validators: QtObject {
+        readonly property list<StatusValidator> keypairName: [
+            StatusValidator {
+                name: "startsWithSpaceValidator"
+                validate: function (t) { return !t.startsWith(" ") }
+                errorMessage: qsTr("Keypair starting with whitespace are not allowed")
+            },
+            StatusRegularExpressionValidator {
+                regularExpression: /^[a-zA-Z0-9\-_ ]+$/
```
Comment (caybro) [inline ui/imports/utils/Constants.qml:482]: `Constants.regularExpressions.alphanumericalExpanded`?

File: ui/imports/utils/Constants.qml [hunk 3]
```diff
@@ -466,7 +466,28 @@ QtObject {
...
+                errorMessage: qsTr("Only letters, numbers, underscores, whitespaces and hyphens allowed")
```
Comment (caybro) [inline ui/imports/utils/Constants.qml:483]: `Constants.errorMessages.alphanumericalExpandedRegExp`?

File: ui/imports/utils/Constants.qml [hunk 4]
```diff
@@ -489,8 +510,8 @@ QtObject {
             // TODO: Create `StatusMaxLengthValidator` in StatusQ
             StatusValidator {
                 name: "maxLengthValidator"
-                validate: function (t) { return t.length <= 24 }
-                errorMessage: qsTr("24 character username limit")
+                validate: function (t) { return t.length <= keypair.nameLengthMax }
+                errorMessage: qsTr("%1 character username limit").arg(keypair.nameLengthMax)
```
Comment (caybro) [inline ui/imports/utils/Constants.qml:514]:
```suggestion
                errorMessage: qsTr("%n character(s) username limit", "", keypair.nameLengthMax)
```

File: ui/imports/utils/Constants.qml (hunk 2 above)
Comment (saledjenic) [reply ui/imports/utils/Constants.qml:482]: `alphanumericalExpanded` is this `/^$|^[a-zA-Z0-9\-_ ]+$/` which allows for an empty string, that's why I didn't use it.

**Comment by saledjenic** [comment]
@caybro thanks for the review, comments applied.

In general I saw constants you mentioned above, but didn't use it, we need to sort out those validators a bit, cause now we have the same validators in some places with inconsistent error messages. My proposal would be to have all possible single validators listed like:

```qml
readonly property QtObject validators: QtObject {
        readonly property StatusValidator cannotStartWithSpace: StatusValidator {
            name: "startsWithSpaceValidator"
            validate: function (t) { return !t.startsWith(" ") }
            errorMessage: qsTr("Keypair starting with whitespace are not allowed")
        }
        
        readonly property StatusValidator lettersNumbersUnderscoresWhitespacesHyphens: StatusRegularExpressionValidator {
            regularExpression: /^[a-zA-Z0-9\-_ ]+$/
            errorMessage: errorMessages.alphanumericalExpandedRegExp
        }
        
        readonly property StatusValidator keypairMinLength: StatusMinLengthValidator {
            minLength: keypair.nameLengthMin
            errorMessage: qsTr("Keypair must be at least %n character(s)", "", keypair.nameLengthMin)
        }
        .
        .
        .
    }
```

Then define validator types where each type will represent the list of validators:
```qml
readonly property QtObject validatorTypes: QtObject {
        readonly property string keypairName: "keypairName"
        readonly property string profileName: "profileName"
        readonly property string communityName: "communityName"
        .
        .
        .
    }
```
And finally define a function which will return a list of validators based on validator type. This function will be used in all places in the app and we will have somehow centralized, single place where validators are coming from.
```qml
function getValidatorsByType(type) {
        switch(type) {
        case Constants.validatorTypes.keypairName:
            return [].append(Constants.validators.cannotStartWithSpace,
                             Constants.validators.lettersNumbersUnderscoresWhitespacesHyphens,
                             Constants.validators.keypairMinLength)
        case Constants.validatorTypes.profileName:
            return [].append(...)
        case Constants.validatorTypes.communityName:
            return [].append(...)
        .
        .
        .
    }
```

File: ui/imports/utils/Constants.qml (hunk 2 above)
Comment (caybro) [reply ui/imports/utils/Constants.qml:482]: Ah right, I see

**Comment by caybro** [comment]
> In general I saw constants you mentioned above, but didn't use, we need to sort out those validators a bit, cause now we have the same validators in some places with inconsistent error messages. My proposal would be to have all possible single validators listed like:

Cool, I like that suggestion, much more concise

**Review by caybro — APPROVED** [review]
LGTM

**Comment by saledjenic** [comment]
@benjthayer @anastasiyaig thanks for spotting these inconsistencies between Figma and the desktop app, but just on last Thursday those labels were as they are in the app, cause when I was implementing them I literally copied them from Figma to appropriate place in code. Seems they were updated in meantime. Will update them within this PR, no worries. :)

**Comment by anastasiyaig** [comment]
figma is alive!

**Comment by saledjenic** [comment]
@anastasiyaig please check it again, hope all should be handled now.

- labels changed in meantime in Figma are updated in the app
- `Show encrypted QR of keypairs on device` is displayed only if needed now,
for this I had to add account/keypair `operability` feature, so far it handles only
displaying a context menu option, later in a separate issue a proper functionality
will be added
- wallet related toast notifications moved to the proper place (`AppMain.qml`, where
actually all notifications should be handled), not in the place where they occurs, we
need to review other notifications as well and move them there

**Review by anastasiyaig — APPROVED** [review]

**Comment by saledjenic** [comment]
@anastasiyaig the task where we will handle that should be syncing - edge cases #10772.

**Pull request merged**